tests/Tagged/TaggedProviderTest.php

```diff
@@ -0,0 +1,121 @@
+<?php
+declare(strict_types=1);
+
+namespace Fig\EventDispatcher\Tagged;
+
+use Fig\EventDispatcher\ParameterDeriverTrait;
+use Fig\EventDispatcher\TaggedProviderTrait;
+use Fig\EventDispatcher\CollectingEvent;
+use PHPUnit\Framework\TestCase;
+use Psr\EventDispatcher\ListenerProviderInterface;
+
+class TaggedProviderTest extends TestCase
+{
+
+    /** @var ListenerProviderInterface */
+    protected $provider;
+
+    public function setUp()
+    {
+        parent::setUp();
+
+        $this->provider = new class implements ListenerProviderInterface {
+            use TaggedProviderTrait;
+            use ParameterDeriverTrait;
+
+            /** @var array */
+            protected $listeners = [];
```

Comment:
Since you cannot extend an anonymous class... why would you ever use `protected`?

Reply:
Habit?  (And in this case, copy/paste.)  I almost never use private.